cmd/relayproxy/config/config.go

```diff
@@ -91,10 +92,14 @@ func New(flagSet *pflag.FlagSet, log *zap.Logger, version string) (*Config, erro
 			log.Error("error loading file", zap.Error(errBindFile))
 		}
 	}
-
+	configMap := k.Raw()
 	// Map environment variables
-	_ = k.Load(env.Provider("", ".", func(s string) string {
-		return strings.ReplaceAll(strings.ToLower(s), "_", ".")
+	_ = k.Load(env.ProviderWithValue("", ".", func(s string, v string) (string, interface{}) {
+		if strings.HasPrefix(s, "RETRIEVERS") || strings.HasPrefix(s, "NOTIFIERS") {
+			_ = loadArrayEnv(s, v, configMap)
```

Comment:
Here we are ignoring the error.
We should at least log the error to be sure that the user has feedback if it does not work as expected.

```suggestion
			err := loadArrayEnv(s, v, configMap)
			log.Error("config: error loading array env", zap.String("key", s), zap.String("value", v), zap.Error(err))
```